rpc/nns/hashes.go

```diff
@@ -36,27 +36,69 @@ func InferHash(sg ContractStateGetter) (util.Uint160, error) {
 	return c.Hash, nil
 }
 
+// NewFSReader creates an instance of [ContractReader] using hash obtained via
+// [InferHash].
+func NewFSReader(sg ContractStateGetter, invoker Invoker) (*ContractReader, error) {
+	h, err := InferHash(sg)
+	if err != nil {
+		return nil, err
+	}
+	return NewReader(invoker, h), nil
+}
+
+// NewFS creates an instance of [Contract] using hash obtained via [InferHash].
+func NewFS(sg ContractStateGetter, actor Actor) (*Contract, error) {
```

Comment:
confusing name to me, i suggest
```suggestion
func NewInferred(sg ContractStateGetter, actor Actor) (*Contract, error) {
```
or
```suggestion
func NewWithInferredHash(sg ContractStateGetter, actor Actor) (*Contract, error) {
```

Reply:
Yeah, I had some doubts about it as well, but decided to try with a shorter one. @carpawell, what'd be your preference?

Reply:
yes, i would die if i did not know what and why FS is here. NewInferred and NewWithInferredHash are not perfect to me either but better

Reply:
Fixed.